assembly/src/tokens/mod.rs

```diff
@@ -307,16 +310,10 @@ fn validate_proc_invocation_label<'a>(
 ///
 /// A label is considered valid if it start with a letter and consists only of letters, numbers,
 /// and underscores.
-fn is_valid_label(label: &str) -> bool {
-    // a label must start with a letter
-    if label.is_empty() || !label.chars().next().unwrap().is_ascii_alphabetic() {
-        return false;
+fn is_valid_label(label: &str) -> bool{
+    match PROCEDURE_LABEL_PARSER.parse_label(label.to_string()) {
+        Ok(_n)  => true,
+        Err(_e) => false,
```

Comment:
Since `parse_label` returns a result, we can use the Rust's convenient `is_ok` function here instead of the `match`

```
    PROCEDURE_LABEL_PARSER.parse_label(label.to_string()).is_ok()
```

Reply:
Sure, though I'm not sure exactly which thing you're asking about. If you're asking why I prefer it, it's because it's more concise & reduces boilerplate, since we can do the same thing in one line instead of 4. If you're asking about the syntax, the docs are here: https://doc.rust-lang.org/std/result/